site/src/pages/DeploySettingsPage/NotificationsPage/NotificationsPage.tsx

```diff
@@ -0,0 +1,268 @@
+import { Link, useSearchParams } from "react-router-dom";
+import {
+  notificationDispatchMethods,
+  selectTemplatesByGroup,
+  systemNotificationTemplates,
+  updateNotificationTemplateMethod,
+} from "api/queries/notifications";
+import { displaySuccess } from "components/GlobalSnackbar/utils";
+import { Loader } from "components/Loader/Loader";
+import { Stack } from "components/Stack/Stack";
+import { TabLink, Tabs, TabsList } from "components/Tabs/Tabs";
+import {
+  castNotificationMethod,
+  methodIcons,
+  methodLabels,
+  type NotificationMethod,
+} from "modules/notifications/utils";
+import { Section } from "pages/UserSettingsPage/Section";
+import { deploymentGroupHasParent } from "utils/deployOptions";
+import { pageTitle } from "utils/page";
+import { useDeploySettings } from "../DeploySettingsLayout";
+import OptionsTable from "../OptionsTable";
+
+type MethodToggleGroupProps = {
+  templateId: string;
+  options: NotificationMethod[];
+  value: NotificationMethod;
+};
+
+const MethodToggleGroup: FC<MethodToggleGroupProps> = ({
+  value,
+  options,
+  templateId,
+}) => {
+  const queryClient = useQueryClient();
+  const updateMethodMutation = useMutation(
+    updateNotificationTemplateMethod(templateId, queryClient),
+  );
+
+  return (
+    <ToggleButtonGroup
+      exclusive
+      value={value}
+      size="small"
+      aria-label="Notification method"
+      css={styles.toggleGroup}
+      onChange={async (_, method) => {
+        await updateMethodMutation.mutateAsync({
+          method,
+        });
+        displaySuccess("Notification method updated");
+      }}
+    >
+      {options.map((method) => {
+        const Icon = methodIcons[method];
+        const label = methodLabels[method];
+        return (
+          <Tooltip key={method} title={label}>
+            <ToggleButton
+              value={method}
+              css={styles.toggleButton}
+              onClick={(e) => {
+                // Retain the value if the user clicks the same button, ensuring
+                // at least one value remains selected.
+                if (method === value) {
+                  e.preventDefault();
+                  e.stopPropagation();
+                  return;
+                }
+              }}
+            >
+              <Icon aria-label={label} />
+            </ToggleButton>
+          </Tooltip>
+        );
+      })}
+    </ToggleButtonGroup>
+  );
+};
+
+export const NotificationsPage: FC = () => {
+  const [searchParams] = useSearchParams();
+  const { deploymentValues } = useDeploySettings();
+  const [templatesByGroup, dispatchMethods] = useQueries({
+    queries: [
+      {
+        ...systemNotificationTemplates(),
+        select: selectTemplatesByGroup,
+      },
+      notificationDispatchMethods(),
+    ],
+  });
+  const ready =
+    templatesByGroup.data && dispatchMethods.data && deploymentValues;
+  const tab = searchParams.get("tab") || "events";
```

Comment:
I don't think it would make too much difference since I'm using links to update the search params. I'm using `||` instead of `??` to grab empty strings as well.